src/mon/AuthMonitor.cc

```diff
@@ -1857,14 +1853,14 @@ bool AuthMonitor::prepare_command(MonOpRequestRef op)
       goto done;
     }
 
-    if (!valid_caps(caps_vec, &ss)) {
+    if (!valid_caps(ceph_caps, &ss)) {
       err = -EINVAL;
       goto done;
     }
 
     map<string,bufferlist> newcaps;
-    for (vector<string>::iterator it = caps_vec.begin();
-	 it != caps_vec.end(); it += 2)
+    for (vector<string>::iterator it = ceph_caps.begin();
```

Comment:
What's going on here? IIUC the `ceph_caps` here is still the same as defined as `map<string, string> ceph_caps` above. If so, it's an instance of `std::map` but the iterator is typed as `vector<string>::iterator`. How does this compile? What I misunderstand?

Reply:
I replaced `caps_vec` by `ceph_caps` for entire file and, I suppose, in this case I eventually forgot to modify the loop.

Reply:
Fixed.

Reply:
Yeah. What's interesting is how this has compiled.

Reply:
>  How does this compile? What I misunderstand?

This compiled because in eventual commits this got was deleted.

Reply:
ah, I see. Thanks!